libs/designsystem/toggle/src/toggle.component.ts

```diff
@@ -29,14 +34,40 @@ import { IonToggle } from '@ionic/angular/standalone';
   ],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class ToggleComponent implements ControlValueAccessor, OnInit {
+export class ToggleComponent implements ControlValueAccessor, OnInit, AfterViewInit {
+  @ViewChild(IonToggle, { read: ElementRef, static: true })
+  private ionToggleElement?: ElementRef<HTMLIonToggleElement>;
+
   constructor(
     private cdr: ChangeDetectorRef,
-    private elementRef: ElementRef<HTMLElement>
+    private elementRef: ElementRef<HTMLElement>,
+    private ionicElementPartHelper: IonicElementPartHelper
   ) {}
 
   ngOnInit(): void {
+    /**
+     * We cannot query ion-toggle for slotted content at this point as the slot has not been rendered.
+     * But we need to know if content is slotted to set justify and labelPlacement BEFORE ion-toggle is rendered.
+     * So it has to be done by querying an additional wrapper around the default content slot like this.
+     */
+    this._hasSlottedContent = this.elementRef.nativeElement
+      .querySelector('.default-content')
+      .hasChildNodes();
+
+    const slot = this.elementRef.nativeElement.getAttribute('slot');
+    if (slot === 'end' && this._hasSlottedContent) {
+      this._justify = 'space-between';
+    }
+
+    if (slot === 'start' && this._hasSlottedContent) {
+      this._labelPlacement = 'end';
+    }
+
     this.inheritAriaAttributes();
+    if (!this._labelText && !this._hasSlottedContent) {
+      // if no label has been set try to find a label in an item and use its text content
+      this._labelText = setAccessibleLabel(this.elementRef.nativeElement);
+    }
```

Comment:
Consider re-using the `inheritAriaLabelText` helper function similar to how we do for checkbox and radio. 

Also consider re-using that whole structure, to completely align how we do for the three components:
```
ngAfterContentInit(): void {
    this._labelText = inheritAriaLabelText(this.element.nativeElement);

    if (!this.text && !this._labelText && !this._hasSlottedContent) {
      // if no label has been set try to find a label in an item and use its text content
      this._labelText = setAccessibleLabel(this.element.nativeElement);
    }
  }
```